### PR TITLE
CFY-6866 Added upload blueprints and create deployments scale tests

### PR DESCRIPTION
**Pull request by inbala**

### CODE REVIEW

File: cosmo_tester/framework/examples/hello_world.py [hunk 1]
```diff
@@ -31,6 +31,7 @@ class HelloWorldExample(AbstractExample):
     def __init__(self, *args, **kwargs):
         super(HelloWorldExample, self).__init__(*args, **kwargs)
         self.disable_iptables = False
+        self.name = 'hello_world'
```
Comment (dusking) [inline cosmo_tester/framework/examples/hello_world.py]: name of what?

File: cosmo_tester/framework/examples/__init__.py [hunk 2]
```diff
@@ -46,6 +46,15 @@ def __init__(self, cfy, manager, attributes, ssh_key, logger, tmpdir,
         self.blueprint_id = 'hello-{0}'.format(str(uuid.uuid4()))
         self.deployment_id = self.blueprint_id
         self.verify_metrics = True
+        self._name = str(uuid.uuid4())
```
Comment (dusking) [inline cosmo_tester/framework/examples/__init__.py]: just use name..

File: cosmo_tester/test_suites/scale_tests/blueprints_and_deployments_scale_test.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
+########
```
Comment (dusking) [inline cosmo_tester/test_suites/scale_tests/blueprints_and_deployments_scale_test.py]: use test_**

File: cosmo_tester/test_suites/scale_tests/blueprints_and_deployments_scale_test.py [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+from manager_rest.storage.models_states import ExecutionState
+from cosmo_tester.framework.fixtures import image_based_manager
+from cosmo_tester.framework.fixtures import hello_world  # noqa
+
+manager = image_based_manager
+DEPLOYMENTS_COUNT = 10
+PROCESS_COUNT = 10
+
+
+def test_hello_world_deployments_scale(             # noqa
```
Comment (dusking) [inline cosmo_tester/test_suites/scale_tests/blueprints_and_deployments_scale_test.py]: maybe test_many_deployments??

File: cosmo_tester/test_suites/scale_tests/blueprints_and_deployments_scale_test.py [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+    hello_world.clone_example()
+    _assert_create_deployments(manager.client, hello_world, logger)
+    _waiting_for_active_executions(manager.client)
+    _assert_create_snapshot(manager.client)
+    _check_response_time(manager.client, logger)
+    _assert_delete_deployments(manager.client, hello_world.name, logger)
+
+
+def _assert_create_deployments(client, blueprint_example, logger):
```
Comment (dusking) [inline cosmo_tester/test_suites/scale_tests/blueprints_and_deployments_scale_test.py]: get DEPLOYMENTS_COUNT, PROCESS_COUNT as input
and some comments about the test..

File: cosmo_tester/test_suites/scale_tests/blueprints_and_deployments_scale_test.py [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+                                        status=ExecutionState.ACTIVE_STATES)
+    return len(executions.items) == 0
+
+
+def _deployments_count(client):
+    deployments_list = client.deployments.list()
+    return deployments_list.metadata.pagination.total
+
+
+def _blueprints_count(client):
```
Comment (dusking) [inline cosmo_tester/test_suites/scale_tests/blueprints_and_deployments_scale_test.py]: _assert_blueprints_count(.., expected_size)

File: cosmo_tester/test_suites/scale_tests/blueprints_and_deployments_scale_test.py [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+        process.join()
+
+    assert _blueprints_count(client) == DEPLOYMENTS_COUNT
+    assert _deployments_count(client) == DEPLOYMENTS_COUNT
+
+
+def _create_deployments(
+        client, blueprint_example, blueprints_count, start_index):
+
+    for i in range(start_index, start_index + blueprints_count):
```
Comment (dusking) [inline cosmo_tester/test_suites/scale_tests/blueprints_and_deployments_scale_test.py]: xrange..

File: cosmo_tester/framework/examples/__init__.py (hunk 2 above)
Comment (dusking) [inline cosmo_tester/framework/examples/__init__.py]: maybe resource_name..

File: cosmo_tester/test_suites/scale_tests/blueprints_and_deployments_scale_test.py [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+def _waiting_for_deployment_execution(client, deployment_id):
+    executions = client.executions.list(deployment_id=deployment_id)
+    return executions[0].status in ['terminated', 'failed']
+
+
+def _assert_create_snapshot(client):
+    snapshot_id = str(uuid.uuid4())
+    client.snapshots.create(snapshot_id, True, True)
+    snapshot_util.assert_snapshot_created(snapshot_id, client)
+    client.snapshots.list()
```
Comment (dusking) [inline cosmo_tester/test_suites/scale_tests/blueprints_and_deployments_scale_test.py]: some comment, why we need the list..

File: cosmo_tester/test_suites/scale_tests/blueprints_and_deployments_scale_test.py [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+    deployments_list = client.deployments.list()
+    return deployments_list.metadata.pagination.total
+
+
+def _blueprints_count(client):
+    blueprints_list = client.blueprints.list()
+    return blueprints_list.metadata.pagination.total
+
+
+def _check_response_time(client, logger):
```
Comment (dusking) [inline cosmo_tester/test_suites/scale_tests/blueprints_and_deployments_scale_test.py]: comment.. currently it's not a real test, just measuring..

File: cosmo_tester/test_suites/scale_tests/blueprints_and_deployments_scale_test.py [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+    assert _blueprints_count(client) == 0
+    assert _deployments_count(client) == 0
+
+
+@retry(stop_max_attempt_number=5,
+       wait_fixed=5000,
+       retry_on_result=lambda r: not r)
+def _waiting_for_deployment_execution(client, deployment_id):
+    executions = client.executions.list(deployment_id=deployment_id)
+    return executions[0].status in ['terminated', 'failed']
```
Comment (dusking) [inline cosmo_tester/test_suites/scale_tests/blueprints_and_deployments_scale_test.py]: remove or use [status in ['terminated', 'failed'] for status in executions]

File: cosmo_tester/test_suites/image_based_tests/hello_world_test.py [hunk 11]
```diff
@@ -13,32 +13,20 @@
 #    * See the License for the specific language governing permissions and
 #    * limitations under the License.
 
-import pytest
-
-from cosmo_tester.framework.examples.hello_world import HelloWorldExample
-from cosmo_tester.framework.fixtures import image_based_manager
+from cosmo_tester.framework.fixtures import image_based_manager, hello_world  # noqa
```
Comment (funkyHat) [inline cosmo_tester/test_suites/image_based_tests/hello_world_test.py]: It shouldn't be necessary to explicitly import fixtures. If you don't then you shouldn't need the other  `# noqa`s either

Comment (inbala) [inline cosmo_tester/test_suites/image_based_tests/hello_world_test.py]: I think it is not necessary only if the fixture is defined in conftest.py .
Anyway, this PR is on hold.

**Pull request closed without merging**